app/domain/util/error_class.rb

```diff
@@ -12,8 +12,9 @@ def initialize(*args)
           @args = args
         end
         define_method(:to_s) do
-          @args.each.with_index.reduce(msg) do |m,(x,i)|
-            m.gsub(Regexp.new("\\{#{i}}"), x || 'nil')
+          @args.each_with_index.reduce(msg) do |m, (x, arg_index)|
```

Comment:
@sgnn7, I'm having trouble groking this.  Can you run through what we're trying to accomplish here?

Reply:
There are multiple problems with the old impl of the class:
- `gsub` cannot auto-convert some types (bools, arrays, fixnums, classes) into strings automatically (error raised)
- `false` was invalidly printed as `nil` since the code does a falsy check
- `each.with_index` is functionally equivalent to `each_with_index` given how we use it

The new code ensures that we use an explicit conversion to string of second gsub param and that we only label 'nil' things that are actually nil and not just falsy.

Try running https://gist.github.com/sgnn7/46dbfbb4bb6d63f527a2b778ddb9315a to see the diff in output